docs/source/guides/auth.mdx

```diff
@@ -0,0 +1,164 @@
+
+The Apollo MCP Server repo comes with an example graph and a Space Devs schema. You'll use this example graph to test the authentication setup.
+
+#### Define authorization and authentication rules in the router
+
+1. In the `graphql/TheSpaceDevs` directory, create a new file called `router.yaml`.
+1. Paste the following configuration, replacing `<AUTH0 DOMAIN>` with your Auth0 domain:
+
+    ```yaml title="Router configuration for JWT authentication"
+    authorization:
+      require_authentication: true # Enforces authentication on all requests
+    authentication:
+      router:
+        jwt:
+          jwks:
+            - url: https://<AUTH0 DOMAIN>/.well-known/jwks.json
+    homepage:
+      enabled: false
+    sandbox:
+      enabled: true
+    supergraph:
+      introspection: true
+    ```
+  
+    This router configuration enforces authentication on all requests and uses the JWKS endpoint provided by Auth0 to validate JWTs:
+
+#### Retrieve your GraphOS license credentials for auth
+
+1. Navigate to [GraphOS Studio](https://studio.apollographql.com/) and log in.
+1. If you have an existing graph, retrieve its graph ref and API key from the **Settings** page.
+1. If you don't have a graph, create a new one by clicking **Add graph** and **Connect an existing graph**. Give it a name and retrieve its graph ref and API key from the modal. You don't need to publish the graph at this point.
+
+1. In the root of the `apollo-mcp-server` directory, run the following command to start the MCP Server and the Router.
+
+    ```sh
+    APOLLO_GRAPH_REF=<YOUR_APOLLO_GRAPH_REF> APOLLO_KEY=<YOUR_APOLLO_KEY> \
+    rover dev --supergraph-config ./graphql/TheSpaceDevs/supergraph.yaml \
+    --router-config ./graphql/TheSpaceDevs/router.yaml \
+    --mcp ./graphql/TheSpaceDevs/config.yaml
+    ```
+
+1. Test the router by navigating to `http://localhost:4000` in your browser. You should see the Explorer.
+
+1. Run the following operation: 
+
+    ```graphql
+    query GetAstronautsCurrentlyInSpace {
+      astronauts(filters: { inSpace: true, search: "" }) {
+        results {
+          id
+          name
+          timeInSpace
+          lastFlight
+          agency {
+            name
+            abbrev
+            country {
+              name
+            }
+          }
+          nationality {
+            name
+            nationalityName
+          }
+          image {
+            thumbnail
+          }
+        }
+      }
+    }
+    ```
+
+1. You should see an `UNAUTHENTICATED` error, which means the router is enforcing authentication. Great! Now you can proceed to test the authentication with an MCP client.
+
+### Make requests with MCP Inspector
+
+1. Start MCP Inspector by running:
+
+    ```sh
+    npx @modelcontextprotocol/inspector
+    ```
+
+    The browser should open automatically with a proxy auth token.
+
+1. In the MCP Inspector, select `Streamable HTTP` as the Transport Type and enter `http://127.0.0.1:5000/mcp` as the URL.
```

Comment:
Is it okay to use `http://127.0.0.1:5000/mcp` here while we had `resource: http://localhost:5000` in the config file above? I'm a bit fixated on this tiny detail because I’ve wasted a lot of time ignoring it before. 😂